packages/a-msgpack/src/Encoder.ts

```diff
@@ -359,7 +359,14 @@ export class Encoder {
   }
 
   encodeExtension(ext: ExtData): void {
-    const size = ext.data.length;
+    this.encodeExtensionHeader(ext);
+    if (ext.data !== WRITE_ONLY_HEADER) {
```

Comment:
also, why not check for length? or maybe even writeU8a already handles the empty data case?